### DISCUSSION TITLE
Install Pip Failed onnxruntime~=1.7

### DISCUSSION THREAD

**Discussion by PRMacDonald** [neuropoly/axondeepseg#714] [Help needed] [answered]
Hi,

I'm having an issue with the installation of ADS. I get an error during the `conda env create` step. It returns with:
 
```
Pip subprocess error:
ERROR: Could not find a version that satisfies the requirement onnxruntime~=1.7 (from ivadomed) (from versions: none)
ERROR: No matching distribution found for onnxruntime~=1.7

failed

CondaEnvException: Pip failed
```

I have been following the steps in the [installation guide](https://axondeepseg.readthedocs.io/en/latest/documentation.html#installation) including installing miniconda. I don't have much experience with the Mac's command line or Macs in general (more comfortable in windows). 

My problem seems similar to the thread "[installation problems MacOS #652](https://github.com/axondeepseg/axondeepseg/discussions/652)" however I was unable to figure this out using the information I found there. 
- I'm not using miniforge
- My Mac doesn't seem to have the option of opening up the terminal with Rosetta

Thanks for any help!

![Mac_specs](https://user-images.githubusercontent.com/57374289/213774735-adea9f9b-3fa6-4af1-b151-87aff3d0d748.png)

**Comment by mathieuboudreau** [comment, marked as answer]
Hi @PRMacDonald - this sounds like a compatibility issue between your computer environment and that software package. I can definitely give you a few tips to debug the source, but if there is something fundamentally off then maybe you'll need to contact that package directly.

You could try `pip install onnxruntime==` to see if there are any available versions at all?

Also verify which python version runs in your conda env (eg run `python --version`).

**Reply by PRMacDonald** [reply]
Hi @mathieuboudreau thanks for the quick response! I tried `pip install onnxruntime==` in the environment but got:

```
msiss-imac-3:axondeepseg dowlinglab$ conda activate ads_venv
(ads_venv) msiss-imac-3:axondeepseg dowlinglab$ pip install onnxruntime==
ERROR: Could not find a version that satisfies the requirement onnxruntime== (from versions: none)
ERROR: No matching distribution found for onnxruntime==
(ads_venv) msiss-imac-3:axondeepseg dowlinglab$ 
```

The python version is 

```
(ads_venv) msiss-imac-3:axondeepseg dowlinglab$ python --version
Python 3.8.15
```

How should I proceed?

**Reply by mathieuboudreau** [reply]
Hi @PRMacDonald  - it seems that there are no onnxruntime versions available at all that are compatible with your current setup – this is strange.

I'd suggest creating an issue on their repository (https://github.com/microsoft/onnxruntime), as someone there will be more experienced and better suited in figuring out why your setup doesn't appear to be compatible with any of their releases.

**Reply by PRMacDonald** [reply]
Okay well thanks anyway, I'll reach out.

Cheers

**Reply by hermancollin** [reply]
Hmm wait I think in this specific case, the command suggested will not work because of the `==` (pip will look for a package called onnxruntime== which doesn't exist). I think what Mathieu meant was `pip install onnxruntime`.

**Reply by Stoyan-I-A** [reply]
> pip will look for a package called onnxruntime== which doesn't exist

It will actually look for onnxruntime but for a version that doesn't exist. The output message should display the available versions. In this user's case, there were none.

But on my machine for example, I'll have the following output:
```
(General_testing2) C:\Users\Stoyan>pip install onnxruntime==
ERROR: Could not find a version that satisfies the requirement onnxruntime== (from versions: 1.7.0, 1.8.0, 1.8.1, 1.9.0, 1.10.0, 1.11.0, 1.11.1, 1.12.0, 1.12.1, 1.13.1)
ERROR: No matching distribution found for onnxruntime==
```

**Reply by mathieuboudreau** [reply]
Yup sorry that I wasnt clear, what @Stoyan-I-A said is why I was asking the user for this info
